File: HRMS/HRMS.Web/Modules/Operations/LeaveApproval/LeaveApprovalForm.cs
using Serenity.ComponentModel;

namespace HRMS.Operations.Forms;

[FormScript("Operations.LeaveApproval")]
[BasedOnRow(typeof(LeaveApprovalRow), CheckNames = true)]
public class LeaveApprovalForm
{
    public int LeaveId { get; set; }
    public int ApproverId { get; set; }
    public int ApprovalLevel { get; set; }
    public DateTime ApprovalDate { get; set; }
    public LeaveStatus Status { get; set; }
    [TextAreaEditor(Rows = 2)]
    public string Remarks { get; set; }
    public bool EscalationTrigger { get; set; }
    public int EscalationTo { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/LeaveApprovalForm.ts
import { BooleanEditor, DateEditor, EnumEditor, initFormType, IntegerEditor, LookupEditor, PrefixedContext, TextAreaEditor } from "@serenity-is/corelib";
import { LeaveStatus } from "./LeaveStatus";

export interface LeaveApprovalForm {
    LeaveId: LookupEditor;
    ApproverId: LookupEditor;
    ApprovalLevel: IntegerEditor;
    ApprovalDate: DateEditor;
    Status: EnumEditor;
    Remarks: TextAreaEditor;
    EscalationTrigger: BooleanEditor;
    EscalationTo: LookupEditor;
}

export class LeaveApprovalForm extends PrefixedContext {
    static readonly formKey = 'Operations.LeaveApproval';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!LeaveApprovalForm.init) {
            LeaveApprovalForm.init = true;

            var w0 = LookupEditor;
            var w1 = IntegerEditor;
            var w2 = DateEditor;
            var w3 = EnumEditor;
            var w4 = TextAreaEditor;
            var w5 = BooleanEditor;

            initFormType(LeaveApprovalForm, [
                'LeaveId', w0,
                'ApproverId', w0,
                'ApprovalLevel', w1,
                'ApprovalDate', w2,
                'Status', w3,
                'Remarks', w4,
                'EscalationTrigger', w5,
                'EscalationTo', w0
            ]);
        }
    }
}

[LeaveStatus]; // referenced types